### PR TITLE
Unit tests and fixes for founds issues for dns/_conf.py

### PR DESCRIPTION
**Pull request by oherrala**
xmas-of-tests branch to pick easiest low hanging fruits (bugs) by adding more unit tests.

### CODE REVIEW

**Comment by oherrala** [comment]
@jviide, I'm done with dns/_conf.py and tests. Please, merge or throw back to me with comments.

File: idiokit/dns/tests/test_dns_conf.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    def setUp(self):
+        missing = tempfile.NamedTemporaryFile()
+        self._missing_file = missing.name
+        missing.close()  # Close deletes the file
+
+        self._empty_file = tempfile.NamedTemporaryFile()
+        self._empty_file.flush()
+
+        # Make sure empty and missing files are not same
+        self.assertNotEqual(self._empty_file.name, self._missing_file)
```
Comment (jviide) [inline idiokit/dns/tests/test_dns_conf.py]: Could you get rid of this assertion just by calling ```missing.close()``` after you have created self._empty_file?

File: idiokit/dns/tests/test_dns_conf.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+             "198.51.100.1 # This line should be ignored\n"
+             "256.256.256.256 invalid-ip # Invalid IP should be ignored\n",
+             "\n",
+             "  198.51.100.126    IPv4.documentation.net.example   \n",
+             "# Hosts file can have tab separator also\n",
+             "2001:DB8::cafe \t IPv6.documentation.net.example\n",
+             "198.51.100.0 incomplete.last.line.net.example"
+             ]
+        )
+        self._hosts.flush()
```
Comment (jviide) [inline idiokit/dns/tests/test_dns_conf.py]: Doing all initialization for all tests in setUp is fine, but would it be clearer to do only the relevant parts inside the test methods themselves? For example having a contextmanager that creates a tempfile along these lines:

```python
@contextlib.contextmanager
def templines(*lines):
    tmp = tempfile.NamedTemporaryFile()
    try:
        tmp.writelines(lines)
        tmp.flush()
        yield tmp.name
    finally:
        tmp.close()
```

Then you could use it inside the test methods:

```python
with templines("# Comments are ignored\n", "198.51.100.0 incomplete.last.line.net.example") as path:
    hosts = _conf.hosts(path=path).load()
    ...
```

Same with missing file tests and so on.

File: idiokit/dns/tests/test_dns_conf.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
+import unittest
+import tempfile
+
+from .. import _conf
+
+
+class HostsFileTests(unittest.TestCase):
+    _missing_file = None
+    _empty_file = None
+    _hosts = None
```
Comment (jviide) [inline idiokit/dns/tests/test_dns_conf.py]: You don't really need to declare these (class) attributes before use. They do serve as documentation of sorts, but see my comments about setUp.

**Comment by jviide** [comment]
Oh right, and could you rename test_dns_conf.py to test_conf.py to keep it in line with the other tests.

**Comment by mseppanen** [comment]
Looks good.

**Pull request merged**